src/modules/discovery/pages/activity/components/ActivityFeedItem.tsx

```diff
@@ -0,0 +1,333 @@
+        {!isFundingTxActivity && <ActivityTitle resource={resource} />}
+        {isRewardActivity && <RewardsInfo reward={resource as ProjectReward} />}
+        <ActivityDescription resource={resource} />
+        {isGoalActivity && (
+          <>
+            <GoalProgressBar goal={resource as ProjectGoal} />
+            <GoalTargetAmount goal={resource as ProjectGoal} />
+          </>
+        )}
+        {isFundingTxActivity && <ContributorInfo resource={resource} />}
+      </VStack>
+    </VStack>
+  )
+}
+
+export default ActivityFeedItem
+
+const ProjectTitle = ({
+  projectTitle,
+  projectImage,
+  projectName,
+}: {
+  projectTitle: string
+  projectImage: string
+  projectName: string
+}) => {
+  return (
+    <HStack as={Link} to={getPath('project', projectName)}>
+      {projectImage && (
+        <Image width={'32px'} height={'32px'} borderRadius={'8px'} src={projectImage} objectFit="cover" />
+      )}
+      <Body size="lg" medium isTruncated maxW={{ base: '250px', lg: '200px' }}>
+        {projectTitle}
+      </Body>
+    </HStack>
+  )
+}
+
+const ActivityDate = ({ date }: { date: string }) => {
+  return (
+    <Body size="md" muted>
+      {getFormattedDate(date)}
+    </Body>
+  )
+}
+
+const ActivityTypeItem = ({ activityType }: { activityType: ActivityType }) => {
+  const { t } = useTranslation()
+
+  switch (activityType) {
+    case ActivityType.ProjectGoalCreated:
+      return (
+        <HStack spacing={1}>
+          <Box mt={-1}>
+            <PiFlagBannerBold size={'20px'} />
+          </Box>
+          <Body size="md">{t('Goal created')}</Body>
+        </HStack>
+      )
+
+    case ActivityType.ProjectGoalReached:
+      return (
+        <HStack spacing={1}>
+          <Box mt={-1}>
+            <PiFlagBannerBold size={'20px'} />
+          </Box>
+          <Body size="md">{t('Goal reached')}</Body>
+        </HStack>
+      )
+
+    case ActivityType.PostPublished:
+      return (
+        <HStack spacing={1}>
+          <PiNewspaper size={'20px'} />
+          <Body size="md">{t('Wrote post')}</Body>
+        </HStack>
+      )
+    case ActivityType.ProjectRewardCreated:
+      return (
+        <HStack>
+          <Box mt={-0.5}>
+            <PiBag size={'20px'} />
+          </Box>
+          <Body size="md">{t('Created reward')}</Body>
+        </HStack>
+      )
+    case ActivityType.ContributionConfirmed:
+      return (
+        <HStack>
+          <Box mt={-0.5}>
+            <PiLightning size={'20px'} />
+          </Box>
+          <Body size="md">{t('Top Contribution')}</Body>
+        </HStack>
+      )
+    default:
+      return <Body size="md">{t('Unknown activity')}</Body>
+  }
+}
+
+const ActivityTitle = ({ resource }: { resource: ActivityResource }) => {
+  if ('title' in resource && typeof resource.title === 'string') {
+    return (
+      <Body size="xl" medium>
+        {resource.title}
+      </Body>
+    )
+  }
+
+  return null
+}
```

Comment:
I split it that way because the Activity item can represent about 5 different resources types and the layout is also slightly different in mobile. I agree there is a lot of components but it allows the main component to be way more readable that it would otherwise be.

Since the resources are all different I need some type checking here otherwise the compiler would complain. 

No it was not written by AI, I'll change the export tho